src/Microsoft.Extensions.Http.Polly/PolicyHttpMessageHandler.cs

```diff
@@ -77,7 +92,13 @@ protected override Task<HttpResponseMessage> SendAsync(HttpRequestMessage reques
                 throw new ArgumentNullException(nameof(request));
             }
 
-            return _policy.ExecuteAsync((ct) => base.SendAsync(request, ct), cancellationToken);
+            var policy = ResolvePolicy(request);
+            return policy.ExecuteAsync((ct) => base.SendAsync(request, ct), cancellationToken);
```

Comment:
Does there need to be a mechanic here to provide a way to create a `Context` which could be passed into the policy, or is that going to be catered for separately/elsewhere? Given that the policy is already being resolved from the request, there could be a hook to resolve a context from the request as well.

Reply:
See https://github.com/aspnet/HttpClientFactory/pull/65

Whomever is the caller can provide a context, or another message handler can do it (since they are also callers).